mocks/consumer.go

```diff
@@ -289,7 +261,13 @@ func (pc *PartitionConsumer) HighWaterMarkOffset() int64 {
 // reasons forthis not to happen. ou can call ExpectMessagesDrainedOnClose so it will
 // verify that the channel is empty on close.
 func (pc *PartitionConsumer) YieldMessage(msg *sarama.ConsumerMessage) {
-	pc.expectations <- &consumerExpectation{Msg: msg}
+	pc.highWaterMarkOffset += 1
```

Comment:
I removed the atomic operations because it's no longer executed in a goroutine... then again, `handleExpectations` uses a for loop anyway, so that alone would not lead to race conditions. So I guess it's not totally clear to me why they were there.

If atomic operations are needed in case someone calls code concurrently, that means that plain assignments are never safe in public APIs.